packages/cdktf-cli/lib/dependencies/dependency-manager.ts

```diff
@@ -0,0 +1,260 @@
+  provider: ${constraint.simplifiedName}
+  version : ${constraint.version || "latest"}
+  language: ${this.targetLanguage}
+  cdktf   : ${this.cdktfVersion}
+`);
+
+    if (this.targetLanguage === Language.GO) {
+      console.log(
+        `There are no pre-built providers published for Go at the moment. See https://github.com/hashicorp/terraform-cdk/issues/723`
+      );
+      return false;
+    }
+
+    const v = await getPrebuiltProviderVersion(constraint, this.cdktfVersion);
+    const exists = v !== null;
+
+    if (exists) {
+      console.log(`Found pre-built provider.`);
+    } else {
+      console.log(
+        `Pre-built provider does not exist for the given constraints.`
+      );
+    }
+
+    return exists;
+  }
+
+  async addPrebuiltProvider(constraint: ProviderConstraint) {
+    logger.debug(
+      `adding pre-built provider ${constraint.source} with version constraint ${constraint.version} for cdktf version ${this.cdktfVersion}`
+    );
+
+    if (this.targetLanguage === Language.GO) {
+      throw Errors.Usage(
+        "There are no pre-built providers published for Go at the moment. See https://github.com/hashicorp/terraform-cdk/issues/723"
+      );
+    }
+
+    const npmPackageName = await getNpmPackageName(constraint);
+
+    if (!npmPackageName) {
+      throw Errors.Usage(
+        `Could not find pre-built provider for ${constraint.source}`
+      );
+    }
+
+    const packageName = this.convertPackageName(npmPackageName);
+    const prebuiltProviderVersion = await getPrebuiltProviderVersion(
+      constraint,
+      this.cdktfVersion
+    );
+    if (!prebuiltProviderVersion) {
+      throw Errors.Usage(
+        `No pre-built provider found for ${constraint.source} with version constraint ${constraint.version} and cdktf version ${this.cdktfVersion}`
+      );
+    }
+
+    const packageVersion = prebuiltProviderVersion;
+
+    await this.packageManager.addPackage(packageName, packageVersion);
+
+    // TODO: more debug logs
+  }
+
+  async addLocalProvider(constraint: ProviderConstraint) {
+    console.log(
+      `Adding local provider ${constraint.source} with version constraint ${constraint.version} to cdktf.json`
+    );
+
+    if (!constraint.version && constraint.isFromTerraformRegistry()) {
+      const v = await getLatestVersion(constraint);
+      if (v) {
+        constraint = new ProviderConstraint(
+          constraint.source,
+          // "1.3.2" -> "~> 1.3"
+          `~> ${v.split(".").slice(0, 2).join(".")}`
+        );
+      }
+    }
+
+    await new CdktfConfigManager().addProvider(constraint);
+  }
+
+  /**
+   * Converts an NPM package name of a pre-built provider package to the name in the target language
+   */
+  private convertPackageName(name: string): string {
+    switch (this.targetLanguage) {
+      case Language.GO:
+        throw new Error("pre-built providers are not supported for Go yet");
```

Comment:
Can we link an issue to upvote here? And shouldn't there be a hint on how to add a provider for go? Lets say you run `cdktf provider add aws` then we find it's a prebuilt one and end up here where it explodes, how would you know you have to run `cdktf provider add aws --force-local`?

Reply:
This error is not supposed to be surfaced to the user. The behaviour you are describing is happening [here](https://github.com/hashicorp/terraform-cdk/blob/8bd931729fc8bc6716e3c750ce7fd5144da026ec/packages/cdktf-cli/lib/dependencies/dependency-manager.ts#L184) and [here](https://github.com/hashicorp/terraform-cdk/blob/8bd931729fc8bc6716e3c750ce7fd5144da026ec/packages/cdktf-cli/lib/dependencies/dependency-manager.ts#L156-L161). Although the first one currently won't be reached, because we have no "force-pre-built" flag, that might directly jump into that method.
Also, you don't have to run with `--force-local` for Go, it will automatically fall back to local after printing the info that Go has no pre-built providers yet.

Reply:
Ah so it's solved on the prebuilt detection level? (then it would be `Errors.Internal` right?)

Reply:
yeah, it could be errors internal – will change it accordingly 👍